### ISSUE TITLE
Source maps

### ISSUE THREAD

**Issue by conradz** [shapesecurity/shift-codegen-js#3]
As far as I can tell, there is currently no way to generate source maps when using this. It would be great if this would support source maps with an API similar to escodegen's.

**Comment by michaelficarra** [comment]
This is now blocked by #2.

**Comment by bakkot** [comment]
I don't think this is blocked by anything now. Given the current API to `parseScriptWithLocation`, the interface should presumably be a third, optional parameter (or another entry point) taking a [Weak]Map from nodes to locations.

## 

Edit: Or possibly (Map, file name) pairs, given that source maps can map multiple input files to one output.
